examples/credit-risk-end-to-end/01_Credit_Risk_Data_Prep.ipynb

```diff
@@ -0,0 +1,1103 @@
+       "own_telephone                      yes                        yes  \n",
+       "foreign_worker                     yes                        yes  \n",
+       "class                              bad                       good  "
+      ]
+     },
+     "execution_count": 10,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# View first records in training data (transposed)\n",
+    "train.head().T"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6825a3ed-2495-4ded-8f5f-1c64041ba5ac",
+   "metadata": {},
+   "source": [
+    "### Feature Engineering"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5f50349a-4d7e-4c3d-b5a7-a6bb5a5718c3",
+   "metadata": {},
+   "source": [
+    "Once data columns have been prepared so that they can be used to train an AI model, it is common to refer to them as \"features\". The process of preparing features is referred to as \"feature engineering\".\n",
+    "\n",
+    "As we are creating a rather simple baseline model, we will ignore the categorical columns in favor of simplicity, noting that we are potentially leaving a lot of useful information out. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "id": "6573e223-5bcf-4e1d-a807-e2dc5ceaf9e2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Filter on numeric feature columns\n",
+    "train_num = train.loc[\n",
+    "    :, \n",
+    "    (train.dtypes == \"int64\")\n",
+    "].astype(float)\n",
+    "test_num = test.loc[\n",
+    "    :,\n",
+    "    (test.dtypes == \"int64\")\n",
+    "].astype(float)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0017646c-e112-4a7f-9704-5c9f547f08d0",
+   "metadata": {},
+   "source": [
+    "We'll also split out the target (label) data, as this generally is recorded separately from the contextual data in real-life, and it is often worked with separately in AI modeling. In the process, let's also engineer `class` into a numerical column, from categorical (\"good\" | \"bad\") to float (1.0 | 0.0)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "id": "cb930659-f6d0-4448-ad1d-f3dbaf6933ef",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Make \"class\" target variable numeric\n",
+    "train_y = (train[\"class\"] == \"good\").astype(float)\n",
+    "test_y = (test[\"class\"] == \"good\").astype(float)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a451c9a3-0390-4d5a-b687-c59f52445eb1",
+   "metadata": {},
+   "source": [
+    "### Additional Data Preparation (For Demonstrating Feast)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "dc4e7653-b118-44c3-ade3-f1b217b112fc",
+   "metadata": {},
+   "source": [
+    "At this point, it's important to bring up that Feast was developed primarily to work with production data. Feast requires datasets to have IDs and timestamps, which it uses in joins.\n",
```

Comment:
Feast requires datasets to have `IDs` and timestamps
May be better to use the word "primary key", ignore my comment if it makes sense to a data engineer, needs a bit more elaboration for new users as feast also supports composite keys. And also why feast mandates it.

Reply:
I added a sentence to explain this.